<launch>
  <arg name="uri" default="radio://0/100/2M/E7E7E7E703" />
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="frame" default="cf3"/>
  <arg name="pose" default="/vrpn_client_node/$(arg frame)/pose"/>

  <!-- Optitrack Connection -->
  <include file="$(find vrpn_client_ros)/launch/optitrack.launch" />
  
  <!-- Joystick -->
  <node name="joy" pkg="joy" type="joy_node" output="screen">
	  <param name="dev" value="$(arg joy_dev)"/>
  </node>	

  <!-- Crazyflie Server-->
  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch" />
  <group ns="crazyflie">
	<include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
		<arg name="uri" value="$(arg uri)" />
		<arg name="tf_prefix" value="crazyflie" />
		<arg name="enable_logging" value="False" />
	</include>

	<node name="pid_controller" pkg="lab" type="pid_controller_tune" output="screen">
		<param name="target" value="goal"/>
		<param name="pose" value="$(arg pose)"/>
		<param name="frequency" value="150"/>
		<rosparam command='load' file="$(find lab)/config/drone4.yaml"/>
	</node>

	<node name="pose" pkg="lab" type="joystick_hover.py" output="screen">
		<param name="name" value="goal"/>
		<param name="rate" value="50"/>
		<param name="z_max" value="0.5"/>
	</node>

	<node name="joystick_button" pkg="lab" type="joystick_button.py" output="screen">
	  <param name="joy_topic" value="/joy"/>
	</node>
   </group>

</launch>

